### DISCUSSION TITLE
hipDeviceAttributeComputeCapabilityMajor

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#5590] [General]
In sycl/plugins/hip/pi_hip.cpp, does HIP have DeviceAttributeComputeCapabilityMajor ? 

```  
  if (getAttribute(device, hipDeviceAttributeComputeCapabilityMajor) >= 6) {
      // compute capability 6.x introduces operations that are atomic with
      // respect to other CPUs and GPUs in the system
      if (value & PI_USM_ACCESS)
        value |= PI_USM_ATOMIC_ACCESS;
      if (value & PI_USM_CONCURRENT_ACCESS)
        value |= PI_USM_CONCURRENT_ATOMIC_ACCESS;
    }
```

**Comment by malixian** [comment]
It may be related to the rocm version.  The rocm version I use is 3.9, this attribute is defined in hip/include/hip/hip_runtime_api.h.

**Reply by zjin-lcf** [reply]
I found the comments here https://github.com/ROCm-Developer-Tools/HIP/blob/develop/include/hip/hip_runtime_api.h. Thanks.

    int major;  ///< Major compute capability.  On HCC, this is an approximation and features may
                ///< differ from CUDA CC.  See the arch feature flags for portable ways to query
                ///< feature caps.
    int minor;  ///< Minor compute capability.  On HCC, this is an approximation and features may
                ///< differ from CUDA CC.  See the arch feature flags for portable ways to query
                ///< feature caps.
